### DISCUSSION TITLE
Linking Errors Cause By 3 lines of Code

### DISCUSSION THREAD

**Discussion by VolcanicGameStudios** [community/community#102072] [Programming Help] [answered]
### Body

So I am making a game engine and I am trying to use the ImGui::Begin("Test") and it keeps causing these linking errors

Error	LNK2019	unresolved external symbol "bool __cdecl ImGui::Begin(char const *,bool *,int)" (?Begin@ImGui@@YA_NPEBDPEA_NH@Z) referenced in function "public: virtual void __cdecl ExampleLayer::OnImGuiRender(void)" (?OnImGuiRender@ExampleLayer@@UEAAXXZ)	Sandbox	C:\Dev\Volcanic\Sandbox\SandboxApp.obj	1

Error	LNK2019	unresolved external symbol "void __cdecl ImGui::End(void)" (?End@ImGui@@YAXXZ) referenced in function "public: virtual void __cdecl ExampleLayer::OnImGuiRender(void)" (?OnImGuiRender@ExampleLayer@@UEAAXXZ)	Sandbox	C:\Dev\Volcanic\Sandbox\SandboxApp.obj	1		

Error	LNK2019	unresolved external symbol "void __cdecl ImGui::Text(char const *,...)" (?Text@ImGui@@YAXPEBDZZ) referenced in function "public: virtual void __cdecl ExampleLayer::OnImGuiRender(void)" (?OnImGuiRender@ExampleLayer@@UEAAXXZ)	Sandbox	C:\Dev\Volcanic\Sandbox\SandboxApp.obj	1		

And the code I am having problems with is

#include <Volcanic.h>

#include "imgui/imgui.h"

class ExampleLayer : public Volcanic::Layer
{
public:
	ExampleLayer()
		: Layer("Example")
	{
	}

	void OnUpdate() override
	{
		if (Volcanic::Input::IsKeyPressed(VC_KEY_TAB))
			VC_TRACE("Tab key is pressed (poll)!");
	}

	virtual void OnImGuiRender() override
	{
		ImGui::Begin("Test");
		ImGui::Text("Hello World");
		ImGui::End();
	}

	void OnEvent(Volcanic::Event& event) override
	{
		if (event.GetEventType() == Volcanic::EventType::KeyPressed)
		{
			Volcanic::KeyPressedEvent& e = (Volcanic::KeyPressedEvent&)event;
			if (e.GetKeyCode() == VC_KEY_TAB)
				VC_TRACE("Tab key is pressed (event)!");
			VC_TRACE("{0}", (char)e.GetKeyCode());
		}
	}

};

class Sandbox : public Volcanic::Application
{
public:
	Sandbox()
	{
		PushLayer(new ExampleLayer());
	}

	~Sandbox()
	{

	}

};

Volcanic::Application* Volcanic::CreateApplication()
{
	return new Sandbox();
}

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Bouhrir** [comment]
Make sure you have included the ImGui source files in your project and they are being compiled along with your other source files. If you are using ImGui as a precompiled library, make sure you have included the library file in your linker settings

**Reply by VolcanicGameStudios** [reply]
how do I do that and Also I am using premake

**Reply by VolcanicGameStudios** [reply]
And I check everything and it should work

**Comment by VolcanicGameStudios** [comment, marked as answer]
I fixed it!

I changed my project to a lib instead of a dll.
